base/util.jl

```diff
@@ -788,28 +793,33 @@ unsafe_crc32c(a, n, crc) = ccall(:jl_crc32c, UInt32, (UInt32, Ptr{UInt8}, Csize_
 crc32c(a::Union{Array{UInt8},FastContiguousSubArray{UInt8,N,<:Array{UInt8}} where N}, crc::UInt32=0x00000000) =
     unsafe_crc32c(a, length(a), crc)
 
-crc32c(buf::IOBuffer, crc::UInt32=0x00000000) = unsafe_crc32c(buf.data, min(buf.size, length(buf.data)), crc)
+crc32c(s::String, crc::UInt32=0x00000000) = unsafe_crc32c(s, sizeof(s), crc)
 
 """
-    crc32c(f::IO, nb::Integer, crc::UInt32=0x00000000)
+    crc32c(io::IO, [nb::Integer,] crc::UInt32=0x00000000)
 
-Read up to `nb` bytes from `f` and return the CRC-32c checksum, optionally
-mixed with a starting `crc` integer.
+Read up to `nb` bytes from `io` and return the CRC-32c checksum, optionally
+mixed with a starting `crc` integer.  If `nb` is not supplied, then
+`io` will be read until the end of the stream.
 """
-function crc32c(f::IO, nb::Integer, crc::UInt32=0x00000000)
+function crc32c(io::IO, nb::Integer, crc::UInt32=0x00000000)
+    nb < 0 && throw(ArgumentError("number of bytes to checksum must be ≥ 0"))
     buf = Array{UInt8}(min(nb, 16384))
-    while !eof(f) && nb > 16384
-        n = readbytes!(f, buf)
+    while !eof(io) && nb > 16384
```

Comment:
It'll be catched by the short version so it won't matter too much but in principle 8192 * 2 and 8192 * 4 are equally bad since neither of them makes full use of the LONG loop.